portfolio/src/main/webapp/script.js

```
@@ -26,3 +26,81 @@ function addRandomGreeting() {
+  let firstImageURL = 'images/basketball.JPG';
+  let firstDescription = 'Playing basketball with the Tennis Intramural Team!';
+  displayNewSlide(firstImageURL, firstDescription);
+}
+
+/* Navigate to the slide at slideShowIndex - 1 */
+function previousSlide() {
+  updateSlideShowIndex('down');
+  let imageURL = 'images/' + images[slideShowIndex];
+  let descriptionText = descriptions[slideShowIndex];
+  displayNewSlide(imageURL, descriptionText);
+}
+
+/* Navigate to the slide at slideShowIndex + 1 */
+function nextSlide() {
+  updateSlideShowIndex('up');
+  let imageURL = 'images/' + images[slideShowIndex];
+  let descriptionText = descriptions[slideShowIndex];
+  displayNewSlide(imageURL, descriptionText);
+}
+
+/* Navigate to a random slide */
+function randomSlide() {
+  updateSlideShowIndex('random');
+  let imageURL = 'images/' + images[slideShowIndex];
+  let descriptionText = descriptions[slideShowIndex];
+  displayNewSlide(imageURL, descriptionText);
+}
+
+/* Display slideshow, with correctly navigated content */
+function displayNewSlide(imageURL, descriptionText) {
+  // display slide/image on slideshow
+  const imgElement = document.createElement('img');
+  imgElement.src = imageURL;
+  const imageContainer = document.getElementById('slideshow-content-container');
+  imageContainer.innerHTML = '';
+  imageContainer.appendChild(imgElement);
+
+  // display corresponding description text
+  const paragraph = document.createElement('p');
+  paragraph.appendChild(document.createTextNode(descriptionText));
+  const descriptionContainer = document.getElementById('slideDescription');
+  descriptionContainer.innerHTML = '';
+  descriptionContainer.appendChild(paragraph);
+  console.log(paragraph);
+}
+
+/* Navigating the Slideshow with an index */
+function updateSlideShowIndex(direction) {
+  if (direction == 'up') {
+    slideShowIndex = (slideShowIndex + 1) % images.length;
+  }
+  else if (direction == 'down') {
+    slideShowIndex = slideShowIndex - 1;
```

nit: make 'up', 'down', 'random', etc. into constants, as it will be easier to find bugs

```js
const DIRECTION_UP = 'up';
// ...
updateSlideShowIndex(DIRECTION_UP);
```

(typos will be surfaced as unrecognized name errors instead of being silently ignored)

---

Fixed!